src/main/java/org/checkstyle/autofix/recipe/RedundantImport.java

```diff
@@ -0,0 +1,123 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.autofix.recipe;
+
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.checkstyle.autofix.PositionHelper;
+import org.checkstyle.autofix.parser.CheckstyleViolation;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.tree.J;
+
+public class RedundantImport extends Recipe {
+
+    private final List<CheckstyleViolation> violations;
+
+    public RedundantImport(List<CheckstyleViolation> violations) {
+        this.violations = violations;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Remove redundant imports";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Remove duplicate imports, java.lang imports, and same package imports";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new RemoveRedundantImportsVisitor();
+    }
+
+    private final class RemoveRedundantImportsVisitor extends JavaIsoVisitor<ExecutionContext> {
+
+        private Path sourcePath;
+
+        @Override
+        public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext ctx) {
+
+            this.sourcePath = cu.getSourcePath().toAbsolutePath();
+
+            final Set<String> seenImports = new HashSet<>();
+            final String currentPackage = getCurrentPackage(cu);
+
+            return cu.withImports(
+                    cu.getImports().stream()
+                            .filter(importStmt -> {
+                                return !isRedundant(importStmt, seenImports,
+                                    currentPackage) || !isAtViolationLocation(importStmt); })
+                            .toList()
+            );
+        }
+
+        private boolean isRedundant(J.Import importStmt,
+                                    Set<String> seenImports, String currentPackage) {
+            final String importName = importStmt.getQualid().toString();
+
+            boolean isRedundant = false;
+
+            if (seenImports.contains(importName)) {
+                isRedundant = true;
+            }
+            else {
+                seenImports.add(importName);
+
+                if (importName.startsWith("java.lang.") && !importStmt.isStatic()) {
```

Comment:
technically `!importStmt.isStatic()` is faster than `importName.startsWith("java.lang.")`, You can switch the two conditions